nncf/experimental/onnx/algorithms/min_max_quantization.py

```diff
@@ -0,0 +1,158 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from nncf.common.hardware.config import HWConfigType
+from nncf.common.hardware.config import HW_CONFIG_TYPE_TARGET_DEVICE_MAP
+from nncf.common.quantization.quantizer_propagation.solver import QuantizerPropagationSolver
+from nncf.common.quantization.structs import QuantizableWeightedLayerNode
+from nncf.common.quantization.structs import QuantizerConfig
+from nncf.common.quantization.structs import QuantizationMode
+from nncf.common.insertion_point_graph import InsertionPointGraph
+
+from nncf.experimental.post_training.algorithms.min_max_quantization import MinMaxQuantization
+from nncf.experimental.post_training.algorithms.min_max_quantization import MinMaxQuantizationParameters
+from nncf.experimental.post_training.compressed_model import CompressedModel
+from nncf.experimental.onnx.graph.transformations.layout import ONNXTransformationLayout
+from nncf.experimental.onnx.graph.metatypes.onnx_ops import GENERAL_WEIGHT_LAYER_METATYPES
+from nncf.experimental.onnx.algorithms.quantization.default_quantization import DEFAULT_ONNX_QUANT_TRAIT_TO_OP_DICT
+from nncf.experimental.onnx.graph.transformations.commands import ONNXQuantizerInsertionCommand
+from nncf.experimental.onnx.engine import ONNXEngine
+
+from nncf.experimental.onnx.statistics.collectors import ONNXMinMaxStatisticCollector
+from nncf.experimental.onnx.graph.model_transformer import ONNXModelTransformer
+
+from nncf.experimental.onnx.hardware.fused_patterns import ONNX_HW_FUSED_PATTERNS
+
+from nncf.experimental.onnx.algorithms.quantization.helper import calculate_activation_quantizer_parameters
+from nncf.experimental.onnx.algorithms.quantization.helper import calculate_weight_quantizer_parameters
+
+from nncf.experimental.onnx.hardware.config import ONNXHWConfig
+
+QUANTIZATION_LAYER_METATYPES = GENERAL_WEIGHT_LAYER_METATYPES
+
+
+class ONNXMinMaxQuantization(MinMaxQuantization):
+
+    def __init__(self, statistics_collector,
+                 parameters: MinMaxQuantizationParameters):
+        super().__init__(statistics_collector, parameters)
+        self._weight_quantizers = []
+        self._activation_quantizers = []
+        self.global_quantizer_constraints = {}
+
+    def generate_stat_collector(self, quantizer_config: QuantizerConfig):
+        if self.range_type == 'min_max':
```

Comment:
What should be in the case when `self.range_type != 'min_max'`? Also, `if` statements don't look like a general solution and this moment should be improved. But not in the case of PoC, I think.

Reply:
I would like to implement two options:

- MinMax
- MeanMinMax

I suggest to raise an Exception if `range_type` doesn't belong to the types above.

Reply:
By the way, POT supports not only `mean/max/min`, but the combination of these parameters and `quantile` options.

Reply:
BTW, before we agreed to move to Enums when defining options It is relevant to any option, aggregator type, target device, etc. It means that we should avoid string options. They are allowed only in configs that become only a part of examples and validation CI and should be translated to the options eventually.

Reply:
Translate string parameters to Enum

Reply:
According to the comment of @KodiaqQ:
@AlexKoff88 what do you think about the current design of aggregator_type. There are two options: 

- MinMax
- MeanMinMax

Should I somehow divide them as it is done in POT?

Reply:
After the discussion with @KodiaqQ and @AlexKoff88 we decided:
1) Make batch_size=1 for now
2) Rename StatisticsCollector to StatisticsAggregator.
3) The interface of StatisticsAggreagator should support two scenarios of statistics aggregation:
      a) Calculating statistics from the output tensor and then aggregating them
      b) Do only aggregation, while the model is modified in the way that it already consists of calculating statistics layers.

All of these were enabled in this PR